pkg/kinflate/transformers/overlay.go

```diff
@@ -59,39 +65,143 @@ func (o *overlayTransformer) Transform(baseResourceMap resource.ResourceCollecti
 		switch {
 		case runtime.IsNotRegisteredError(err):
 			// Use JSON merge patch to handle types w/o schema
-			baseBytes, err := json.Marshal(base.Data)
+			merged, err = jsonMergePatch(base.Data.Object, overlay.Data.Object, jsonpatch.MergePatch)
```

Comment:
When merging the patches in the overlay, you called conflict detection and conflict locating functions before `jsonMergePatch`. You may want to do the same here. If there is a conflict, it will fail and the error message may not be very descriptive.

Reply:
It sounds like you are talking about detecting conflict when applying patch to base.
That is a new feature that is out of the scope of this PR. It just override for now.